cmd/assetsvc/postgres_db_test.go

```diff
@@ -0,0 +1,221 @@
+	if err != nil {
+		t.Fatalf("%+v", err)
+	}
+
+	return pam, cleanup
+}
+
+func countTable(t *testing.T, pam *dbutils.PostgresAssetManager, table string) int {
+	var count int
+	err := pam.DB.QueryRow(fmt.Sprintf("SELECT COUNT(*) FROM %s", table)).Scan(&count)
+	if err != nil {
+		t.Fatalf("%+v", err)
+	}
+	return count
+}
+
+func insertCharts(t *testing.T, pam *postgresAssetManager, charts []models.Chart, repo models.Repo) {
+	_, err := pam.EnsureRepoExists(repo.Namespace, repo.Name)
+	if err != nil {
+		t.Fatalf("%+v", err)
+	}
+	for _, chart := range charts {
+		d, err := json.Marshal(chart)
+		if err != nil {
+			t.Fatalf("%+v", err)
+		}
+		_, err = pam.GetDB().Exec(fmt.Sprintf(`INSERT INTO %s (repo_namespace, repo_name, chart_id, info)
+		VALUES ($1, $2, $3, $4)`, dbutils.ChartTable), repo.Namespace, repo.Name, chart.ID, string(d))
+		if err != nil {
+			t.Fatalf("%+v", err)
+		}
+	}
+}
+
+func TestGetChart(t *testing.T) {
+	dbutilstest.SkipIfNoPostgres(t)
+	const repoName = "repo-name"
+
+	testCases := []struct {
+		name string
+		// existingCharts is a map of charts per namespace
+		existingCharts map[string][]models.Chart
+		chartId        string
+		namespace      string
+		expectedChart  string
+		expectedErr    error
+	}{
+		{
+			name:        "it returns an error if the chart does not exist",
+			chartId:     "doesnt-exist-1",
+			namespace:   "doesnt-exist",
+			expectedErr: sql.ErrNoRows,
+		},
+		{
+			name: "it returns the chart matching the chartid",
+			existingCharts: map[string][]models.Chart{
+				"namespace-1": []models.Chart{
+					models.Chart{ID: "chart-1", Name: "my-chart"},
+				},
+			},
+			chartId:       "chart-1",
+			namespace:     "namespace-1",
+			expectedErr:   nil,
+			expectedChart: "my-chart",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			pam, cleanup := getInitializedManager(t)
+			defer cleanup()
+			for namespace, charts := range tc.existingCharts {
+				insertCharts(t, pam, charts, models.Repo{Name: repoName, Namespace: namespace})
+			}
+
+			chart, err := pam.getChart(tc.chartId)
+
+			if got, want := err, tc.expectedErr; got != want {
+				t.Fatalf("got: %+v, want: %+v", got, want)
+			}
+			if got, want := chart.Name, tc.expectedChart; got != want {
+				t.Errorf("got: %q, want: %q", got, want)
+			}
+		})
+	}
+}
+
+func TestGetVersion(t *testing.T) {
+	dbutilstest.SkipIfNoPostgres(t)
+	const repoName = "repo-name"
+
+	testCases := []struct {
+		name string
+		// existingCharts is a map of charts per namespace
+		existingCharts   map[string][]models.Chart
+		chartId          string
+		namespace        string
+		requestedVersion string
+		expectedVersion  string
+		expectedErr      error
+	}{
+		{
+			name:        "it returns an error if the chart does not exist",
+			chartId:     "doesnt-exist-1",
+			namespace:   "doesnt-exist",
+			expectedErr: sql.ErrNoRows,
+		},
+		{
+			name: "it does not ?! return an error if the chart version does not exist",
```

Comment:
This seems like a bug, it should indeed return a failure if the version doesn't exists

Reply:
Updated to expect an error here.